crates/vault-registry/src/lib.rs

```diff
@@ -156,29 +157,32 @@ decl_module! {
         fn on_runtime_upgrade() -> Weight {
             use frame_support::{migration::StorageKeyIterator, Blake2_128Concat};
 
-            if matches!(Self::storage_version(), Version::V0 | Version::V1) {
-                StorageKeyIterator::<T::AccountId, VaultV1<T::AccountId, T::BlockNumber, PolkaBTC<T>, DOT<T>>, Blake2_128Concat>::new(<Vaults<T>>::module_prefix(), b"Vaults")
+            if matches!(Self::storage_version(), Version::V2) {
```

Comment:
Note that we haven't yet run the V2 storage migration, so we should bundle this upgrade into V2.